plugins/modules/packaging/os/pacman_key.py

```diff
@@ -0,0 +1,334 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, George Rawlinson <george@rawlinson.net.nz>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: pacman_key
+author:
+- George Rawlinson (@grawlinson) <george@rawlinson.net.nz>
+version_added: "2.10"
+short_description: Manage pacman's list of trusted keys
+description:
+- Add or remove gpg keys from the pacman keyring.
+notes:
+- Use full key id (16 characters) and fingerprint (40 characters) to avoid key collisions.
+- If you specify both the key id and the URL with C(state=present), the task can verify or add the key as needed.
+- By default, keys will be locally signed after being imported into the keyring.
+- If the specified key id exists in the keyring, the key will not be added.
+requirements:
+- gpg
+- pacman-key
+options:
+    id:
+        description:
+            - The 16 character identifier of the key.
+            - Including this allows check mode to correctly report the changed state.
+            - Do not specify a subkey id, instead specify the master key id.
+        required: yes
+        type: str
+    data:
+        description:
+            - The keyfile contents to add to the keyring.
+            - Must be of "PGP PUBLIC KEY BLOCK" type.
+        type: str
+    file:
+        description:
+            - The path to a keyfile on the remote server to add to the keyring.
+            - Remote file should be of "PGP PUBLIC KEY BLOCK" type.
+        type: path
```

Comment:
I guess `data`, `file` and `url` are mutually exclusive? If yes, that should be mentioned in the documentation.